src/onchain_reading.js

```diff
@@ -8,15 +8,15 @@ const { decodeOrdersBN } = require("./encoding")
 const getOpenOrdersPaginated = async function* (contract, pageSize) {
   let nextPageUser = "0x0000000000000000000000000000000000000000"
   let nextPageUserOffset = 0
-  let lastPageSize = pageSize
+  let hasNextPage = true
 
-  while (lastPageSize == pageSize) {
+  while (hasNextPage) {
```

Comment:
I think in general this flag is an improvement! Definitely makes this a bit more readable.